### PR TITLE
[Evaluation] Bang 'Term's in 'CekValue'

### PR DESCRIPTION
**Pull request by effectfully**
I don't see why these shouldn't be strict given that they all come from the strict `Term`.

### CODE REVIEW

File: plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs [hunk 1]
```diff
@@ -188,8 +188,8 @@ instance Show (BuiltinRuntime (CekValue uni fun)) where
 data CekValue uni fun =
     -- This bang gave us a 1-2% speed-up at the time of writing.
     VCon !(Some (ValueOf uni))
-  | VDelay (Term NamedDeBruijn uni fun ()) !(CekValEnv uni fun)
-  | VLamAbs NamedDeBruijn (Term NamedDeBruijn uni fun ()) !(CekValEnv uni fun)
+  | VDelay !(Term NamedDeBruijn uni fun ()) !(CekValEnv uni fun)
+  | VLamAbs !NamedDeBruijn !(Term NamedDeBruijn uni fun ()) !(CekValEnv uni fun)
```
Comment (michaelpj) [inline plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs:192]: The first bang might be making GHC unpack `NamedDeBruijn`, which isn't necessarily a win. Not sure.

Comment (effectfully) [reply plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs:192]: We don't enable `unbox-strict-fields` globally, right? And it's not enabled by default (unlike `unbox-small-strict-fields`). So I think that shouldn't happen? Gonna rerun the benchmarks.

Comment (effectfully) [reply plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs:192]: Wait, `NamedDeBruijn` isn't even strict? I see neither bangs no `StrictData` in [this](https://github.com/input-output-hk/plutus/blob/master/plutus-core/plutus-core/src/PlutusCore/DeBruijn/Internal.hs) file.

Comment (michaelpj) [reply plutus-core/untyped-plutus-core/src/UntypedPlutusCore/Evaluation/Machine/Cek/Internal.hs:192]: I guess I'm just making things up then. I thought `unbox-small-strict-fields` might hit it too.

**Review by michaelpj — APPROVED** [review]
Fine with me, has conflicts.

**Pull request merged**